lib/rspec/mocks/error_generator.rb

```diff
@@ -142,7 +142,7 @@ def expected_part_of_expectation_error(expected_received_count, expectation_coun
       def actual_method_call_args_description(count, args)
         method_call_args_description(args) ||
           if count > 0 && args.length > 0
-            " with arguments: #{args.inspect.gsub(/\A\[(.+)\]\z/, '(\1)')}"
+            " with arguments: #{format_args(*args)}"
           else
             ""
           end
```

Comment:
Playing around with this now. Does it make sense for one to depend on the received count but not the other? Introducing `count` to both will squash messages like `expected: 0 times with arguments: (\"unexpected text\")\n`. I'm assuming that would not be desirable?